test/custom-task-ctrls/wait-task/pkg/reconciler/reconciler.go

```diff
@@ -67,27 +74,60 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, r *v1alpha1.Run) kreconc
 		return nil
 	}
 
-	dur, err := time.ParseDuration(expr.Value.StringVal)
-	if err != nil {
-		r.Status.MarkRunFailed("InvalidDuration", "The duration param was invalid: %v", err)
+	// Skip if the Run is cancelled.
+	if r.IsCancelled() {
+		logger.Infof("The Custom Task Run %v has been cancelled", r.GetName())
+		r.Status.CompletionTime = &metav1.Time{Time: c.Clock.Now()}
+		var msg string = fmt.Sprint(r.Spec.StatusMessage)
+		if msg == "" {
+			msg = "The Wait Task is cancelled"
+		}
+		r.Status.MarkRunFailed("Cancelled", msg)
 		return nil
 	}
 
-	if r.Status.StartTime == nil {
-		now := metav1.Now()
-		r.Status.StartTime = &now
+	if !r.HasStarted() {
+		logger.Info("Run hasn't started, start it")
+		r.Status.InitializeConditions()
+		r.Status.StartTime = &metav1.Time{Time: c.Clock.Now()}
 		r.Status.MarkRunRunning("Running", "Waiting for duration to elapse")
 	}
 
-	done := r.Status.StartTime.Time.Add(dur)
+	dur, err := time.ParseDuration(expr.Value.StringVal)
+	if err != nil {
+		r.Status.MarkRunFailed("InvalidDuration", "The duration param was invalid: %v", err)
+		return nil
+	}
+	to := r.GetTimeout()
```

Comment:
"The basic rule: the further from its declaration that a name is used, the more descriptive the name must be." - https://github.com/golang/go/wiki/CodeReviewComments#variable-names

`to` is used far from its declaration so it's better to call it `timeout`

```suggestion
	timeout := r.GetTimeout()
```

Reply:
SG, updated!